src/argilla/server/schemas/v1/datasets.py

```diff
@@ -299,11 +310,21 @@ class Record(BaseModel):
 
     class Config:
         orm_mode = True
+        getter_dict = RecordGetterDict
+
+    @classmethod
+    def from_orm(cls: Type["Record"], obj: Any) -> "Record":
+        dict_copy = obj.__dict__.copy()
+        return cls(**{"metadata": dict_copy.pop("metadata_", None), **dict_copy})
```

Comment:
why do we need this?

Reply:
In the listing records endpoints we were using `record.__dict__` to build the `Record` schema. This is because if we use `record` directly, then `pydantic` when creating the `Record` class will try to access `record.responses` and `sqlalchemy` will return an empty list no matter what, so in the output payload we would have `"responses": []` even if the client has not asked for the responses.

To fix this, we use `record.__dict__` which does not include the `responses` if they were not included in the query with a `joinedload`. Now that we've added the `metadata_` attribute, we cannot use `record.__dict__` directly to build the `Record` class, because the field in this class is called `metadata` and we have to map this manually.

As we had to do this in several endpoints, I've decided that it was a good idea to put this logic in this method.

Reply:
Can we use an alias then for the record name? Or use a RecordGetter class to tackle this? I feel this from_orm override is a bit tricky

Reply:
I managed to do the same with the  `GetterDict` and keep the `response` as unset if the query didn't have the `joinedload` option. It was as easy as returning the `default` argument instead of `None`. This way `pydantic` detects that the field has not been set and in combination with the decorator parameter `response_model_exclude_unset=True` it works.